sdk/containerregistry/container-registry/review/container-registry.api.md

```diff
@@ -275,17 +270,19 @@ export interface OciDescriptor {
     annotations?: OciAnnotations;
     digest: string;
     mediaType: string;
-    sizeInBytes: number;
+    size: number;
```

Comment:
Curious - why change to a different name than other languages?

Reply:
I've updated these types so that they match the OCI Image Manifest specification exactly. Given the particularly close relationship between JavaScript objects and JSON, I think keeping the property names and their types as close to the specification as possible makes a lot of sense in TypeScript in this case:
* Doing this allows us to parse manifests of _all_ media types (even those we don't have models for yet) into a weakly typed `manifest` property on `GetManifestResult`, providing greater flexibility for customers. For strong typing, users can call the `isOciImageManifest` type guard; if that function returns true they will now have access to the strong typing of `OciImageManifest`. See [this sample](https://github.com/Azure/azure-sdk-for-js/blob/8be47f33adb93976d391c5898ebfe368def7c9da/sdk/containerregistry/container-registry/samples-dev/downloadImage.ts#L35-L37) for an example. Adding support for additional media types now becomes a simple matter of adding a new model and a new type guard (users can even roll these themselves if they want to bridge the gap).
* I can easily imagine a scenario like this occurring in JavaScript:
  - The user wants to upload a manifest from a file, but they want to check a certain property of the manifest before they upload it.
  - To do this, they read the file and parse its contents using `JSON.parse`. This function does not rename any properties, so the name of the `size` property stays as is, following the spec.
  - They check the property and then try to pass the parsed object to `setManifest`, seeing that it takes an `OciImageManifest` as a parameter. But doing that will cause an error, due to the mismatch of the property names. If I was a user, I would find this behavior quite unintuitive.

To be clear, I don't think that those reasons apply to Java and .NET _at all_, given their strongly typed nature. But to me, this new design feels far more idiomatic for TypeScript than what we had previously.

FWIW, Python is no longer remapping the property names either, as of [this PR](https://github.com/Azure/azure-sdk-for-python/pull/29552/files#diff-41ef66838d6762f64bb9b74b3371daeb1430f2c7e4d95c7bb6ae28d1d46b2327R865), instead representing manifests as a dictionary. I defer to @YalinLi0312 and co. for their reasoning for this change.

Reply:
Thanks for the detailed explanation @timovv! I agree with the rationale and given the nature of JS dropping the suffixes is a good trade-off for the increased flexibility, also making the typeguards available users can make objects strongly typed when they need to.